pkg/test/framework/components/istio/istio.go

```diff
@@ -31,8 +31,48 @@ type Instance interface {
 // SetupConfigFn is a setup function that specifies the overrides of the configuration to deploy Istio.
 type SetupConfigFn func(cfg *Config)
 
+func GetDeployedNamespaces(cfg Config) []string {
+	// Store in map to dedupe
+	allNamespace := map[string]struct{}{}
+
+	if cfg.AlphaInstaller {
+		for _, c := range cfg.InstallComponents {
+			switch c {
+			case Base:
+				allNamespace[cfg.IstioNamespace] = struct{}{}
+				allNamespace[cfg.ConfigNamespace] = struct{}{}
+			case Ingress:
+				allNamespace[cfg.IngressNamespace] = struct{}{}
+			case Egress:
+				allNamespace[cfg.EgressNamespace] = struct{}{}
+			case Telemetry:
+				allNamespace[cfg.TelemetryNamespace] = struct{}{}
+			case Policy:
+				allNamespace[cfg.PolicyNamespace] = struct{}{}
+			case Tracing:
+				allNamespace[cfg.TelemetryNamespace] = struct{}{}
+			case Nodeagent:
+				allNamespace[cfg.IstioNamespace] = struct{}{}
+			}
```

Comment:
does it make sense to add a default: panic("unreachable") to protect against new values somebody forgot to handle here?